openhantek/src/glscope.cpp

```diff
@@ -38,7 +38,7 @@
 /// \brief Initializes the scope widget.
 /// \param settings The settings that should be used.
 /// \param parent The parent widget.
-GlScope::GlScope(DsoSettings *settings, QWidget *parent) : QGLWidget(parent) {
+GlScope::GlScope(DsoSettings *settings, QWidget *parent) : QOpenGLWidget(parent) {
```

Comment:
Qt 5.2 is only 1 year old: https://wiki.qt.io/Qt_5.2_Release
We should support it. My idea would be that you keep your changes and include an own QOpenGLWidget adapter class that inherits from QGLWidget and where you add the missing methods signatures. I will add a buildsystem logic into cmake, so that this adapter class files are only used if Qt is < 5.4. How does that sound?

Reply:
Qt 5.2 is exactly 4 years old: Release notes December 12th, 2013 (well, 5.2.1 has been released 3 months later).

Anyway, QOpenGLWidget is not a QGLWidget subclass, please refer to section __Relation to QGLWidget__ in QOpenGLWidget doc http://doc.qt.io/qt-5/qopenglwidget.html#details .

The adapter can be implemented as alias:
```
#include <QtGlobal>
#if (QT_VERSION >= QT_VERSION_CHECK(5, 4, 0))
#include <QOpenGLWidget>
using GL_WIDGET_CLASS = QOpenGLWidget;
#else
#include <QGLWidget>
using GL_WIDGET_CLASS = QGLWidget;
#endif
```

Please adjust cmake accordingly, I will upload new commit to this PR pretty soon.